x-pack/plugins/index_management/public/plugin.ts

```diff
@@ -81,9 +107,47 @@ export class IndexMgmtUIPlugin {
     };
   }
 
-  public start() {
+  public start(coreStart: CoreStart, plugins: StartDependencies): IndexManagementPluginStart {
+    const { fleet, usageCollection, cloud, share, console } = plugins;
+
     return {
       extensionsService: this.extensionsService.setup(),
+      getIndexMappingComponent: (deps: { history: ScopedHistory<unknown> }) => {
```

Comment:
Creating a getter for this component is generally fine... infinitely better than supplying the component itself.